common/dbtool/dbtool

```diff
@@ -0,0 +1,399 @@
+        return "db/changelog/changelog-update.xml"
+    def default_db_name(self):
+        return "candlepin"
+    def default_user(self):
+        return self.default_db_name();  
+            
+class Gutterball():
+    def classpath(self):
+        return classpath_for_war("gutterball")
+    def create_changelog_location(self):
+        return "db/changelog/changelog.xml";
+    def update_changelog_location(self):
+        return self.create_changelog_location();
+    def default_db_name(self):
+        return "gutterball"
+    def default_user(self):
+        return self.default_db_name();      
+    
+def build_cp(*args):
+    
+    non_empty_args = []
+    for arg in args: 
+        if arg != None and arg != "":
+            non_empty_args+=[arg];
+            
+    if len(non_empty_args)==0:
+        return None
+    return ":".join(non_empty_args)
+             
+               
+def main(cmd_args):
+    parser = OptionParser()
+    
+    action_list = ["create","create-schema-only", "update", "drop-create"]
+    parser.add_option("--action", 
+                      dest="action",
+                      choices=action_list,
+                      help="Action to take on the database. Available actions: %s" % ", ".join(action_list))
+  
+    parser.add_option("-u",
+            dest="dbuser",
+            help="database user to use to create schema and load data. Defaults to name of the product specified by --product")
+    
+    parser.add_option("-d", 
+            dest="dbname",
+            help="database name to use")
+    
+    parser.add_option("-p", 
+            dest="dbpassword",
+            help="password to use for the dbuser")
+    
+    parser.add_option("--classpath", 
+            dest="classpath",
+            help="additional classpath for Liquibase. This is useful for testing purposes or as a workaround")
+    
+    group = OptionGroup(parser, "Postgres specific options")
+
+    group.add_option("--pg-dbhost",
+            dest="postgres_dbhost",
+            help="The database host to use (optional)")
+    
+    group.add_option("--pg-dbport",
+            dest="postgres_dbport",
+            help="The database port to use (optional)")
+    
+    parser.add_option_group(group)
+  
+    
+    parser.add_option("--community",
+            action="store_true", default=False,
+            dest="community",
+            help="true if used in a community fashion")
+    
+    parser.add_option("--database-type",
+            type="choice",
+            default="Postgres",
+            choices=["Postgres", "Oracle", "Mysql"],
+            dest="database_type",
+            help="Type of the database to run against. Valid values are: Postgres, Mysql, Oracle. Defaults to Postgres")
+    
+    parser.add_option("--product",
+            dest="product",
+            type="choice",
+            choices=["candlepin","gutterball"],
+            help="Product. Valid values are: candlepin, gutterball.")
+        
```

Comment:
Depending on whether or not this is going to replace cpdb entirely, we may want to default this to "candlepin" so it can be a drop-in replacement for cpdb without breaking any existing tooling that may use it.